common/cloud_builder.py

```diff
@@ -0,0 +1,258 @@
+"""CloudBuilder executes agents in Cloud Builds."""
+import argparse
+import logging
+import os
+import subprocess
+import time
+import uuid
+from typing import Any
+
+import google.api_core.client_options
+import googleapiclient.errors
+from google.api_core.exceptions import NotFound
+from google.auth import default
+from google.cloud import storage
+from google.oauth2.service_account import Credentials
+from googleapiclient.discovery import build as cloud_build
+
+import utils
+from agent.base_agent import BaseAgent
+from results import BuildResult, Result
+
+OFG_REPO = 'https://github.com/google/oss-fuzz-gen.git'
+OF_REPO = 'https://github.com/google/oss-fuzz.git'
+US_CENTRAL_CLIENT_OPTIONS = google.api_core.client_options.ClientOptions(
+    api_endpoint='https://us-central1-cloudbuild.googleapis.com/')
+
+
+class CloudBuilder:
+  """Encapsulate functions to request Google Cloud Builds to execute agents."""
```

Comment:
nit: we should put a more descriptive text here, and talk about what this `CloudBuilder` represents, rather than about how this is encapsulating functions.

Reply:
+1

Reply:
Added.